woocommerce/api/class-sv-wc-api-base.php

```diff
@@ -99,21 +103,34 @@ protected function perform_request( $request ) {
 
 		$start_time = microtime( true );
 
-		// if this API requires TLS v1.2, force it
-		if ( $this->get_plugin()->require_tls_1_2() ) {
-			add_action( 'http_api_curl', array( $this, 'set_tls_1_2_request' ), 10, 3 );
-		}
+		if ( $this->is_request_cacheable() && ! $this->get_request()->should_refresh() && $response = $this->load_response_from_cache() ) {
+
+			$this->response_loaded_from_cache = true;
 
-		// perform the request
-		$response = $this->do_remote_request( $this->get_request_uri(), $this->get_request_args() );
+		} else {
+
+			// if this API requires TLS v1.2, force it
+			if ( $this->get_plugin()->require_tls_1_2() ) {
+				add_action( 'http_api_curl', array( $this, 'set_tls_1_2_request' ), 10, 3 );
+			}
+
+			// perform the request
+			$response = $this->do_remote_request( $this->get_request_uri(), $this->get_request_args() );
+		}
 
 		// calculate request duration
 		$this->request_duration = round( microtime( true ) - $start_time, 5 );
 
 		try {
 
 			// parse & validate response
-			$response = $this->handle_response( $response );
+			$parsed_response = $this->handle_response( $response );
+
+			// cache the response
+			if ( ! $this->is_response_loaded_from_cache() && $this->is_request_cacheable() ) {
+
+				$this->save_response_to_cache( $response );
+			}
```

Comment:
Perhaps tuck this at the end of `handle_response()` method instead? Could help cut down on the number of conditionals in this method.